africanus/gridding/perleypolyhedron/kernels.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+from numba import prange
+
+try:
+    from scipy.special import jn
+except ImportError as e:
+    # https://stackoverflow.com/a/29268974/1611416, pep 3110 and 344
+    scipy_import_error = e
+else:
+    scipy_import_error = None
+
+from africanus.util.numba import jit, register_jitable
+from africanus.util.requirements import requires_optional
+
+
+@register_jitable
+def uspace(W, oversample):
+    """
+    Generates a kernel sampling of the form
+    0   1   2   3   4
+    |...|...|...|...|...
+    |<----->|
+    half-support (W / 2)
+    | x W
+    . x (oversample - 1) x W
+    where W is odd
+    plus padding by 1 unit on either side
+    """
+    # must be odd so that the taps can be centred at the origin
+    assert W % 2 == 1
+    taps = np.arange(
+        oversample *
+        (W + 2)) / float(oversample) - (W + 2) // 2
+    # (|+.) * W centred at 0
+    return taps
+
+
+def sinc(W, oversample=5, a=1.0):
+    """
+    Basic oversampled sinc window
+    """
+    u = uspace(W, oversample)
+    res = np.sinc(u * a)
+    return res / np.sum(res)
+
+
+@requires_optional('scipy', scipy_import_error)
+def kbsinc(W, b=None, oversample=5, order=15):
+    """
+    Modified keiser bessel windowed sinc (Jackson et al.,
+    IEEE transactions on medical imaging, 1991)
+    with a modification of higher order bessels as default, as
+    this improves the kernel at low number of taps.
+    """
+    autocoeffs = np.polyfit(
+        [1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0],
+        [1.9980, 2.3934, 3.3800, 4.2054, 4.9107, 5.7567, 6.6291, 7.4302], 1)
+    if b is None:
+        b = np.poly1d(autocoeffs)((W + 2))
+
+    u = uspace(W, oversample)
+    wnd = jn(order, b * np.sqrt(1 - (2 * u /
+                                     ((W + 2) + 1))**2)) * 1 / ((W + 2) + 1)
+    res = sinc(W, oversample=oversample) * wnd * np.sum(wnd)
+    return res / np.sum(res)
+
+
+def hanningsinc(W, a=0.5, oversample=5):
```

Comment:
Shouldn't None be the default value for ``a``?

```suggestion
def hanningsinc(W, a=None, oversample=5):
```

Reply:
well spotted yes will fix